### PR TITLE
Bump CMake minimum to 2.8.12

### LINKED ISSUE #3040: CMake 3.19.0 throws deprecation warning in googletest/googletest/CMakeLists.txt:56
```
CMake Deprecation Warning at googletest/googletest/CMakeLists.txt:56 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

### PR DESCRIPTION
**Pull request by chuckatkins**
Fixes #3040

### CODE REVIEW

**Comment by chuckatkins** [comment]
From what I can tell the AppVeyor failure is unrelated.

**Review by atherkhan — APPROVED** [review]

**Comment by acxz** [comment]
It has been a month since this PR has been submitted. @derekmauro can you take a quick look at it?

**Review by Robadob — APPROVED** [review]
Would appreciate this being merged, our CI has just automatically updated CMake version and hit failure due to this.

**Comment by ybgunjal2000** [comment]
unsubscribe

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Note: CMake support is community-based. The maintainers do not use CMake
 # internally.
 
-cmake_minimum_required(VERSION 2.8.8)
+cmake_minimum_required(VERSION 2.8.12)
```
Comment (hjmjohnson) [inline CMakeLists.txt:4]:
```cmake
# Adding ... to version rage to future proof code when using cmake > 3.18 that explicitly 
# deprecated 2.8.12 features
cmake_minimum_required(VERSION 2.8.12...)
```

**Review by MBNwaiser — CHANGES_REQUESTED** [review]
Tell the try

File: CMakeLists.txt (hunk 1 above)
Comment (hjmjohnson) [reply CMakeLists.txt:4]: https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html

**Comment by Be-ing** [comment]
Ping. Lots of projects are stumbling on this bug on CI.

File: CMakeLists.txt (hunk 1 above)
Comment (friendlyanon) [reply CMakeLists.txt:4]: Version range was added in 3.12 to this command and the goal is to support 2.8.12 as a minimum.

Also, empty max range?

Comment (remyjette) [reply CMakeLists.txt:4]: From the linked docs:

> The optional <max> version, if specified, must be at least the <min> version and affects policy settings as described below. If the running version of CMake is older than 3.12, the extra ... dots will be seen as version component separators, resulting in the ...<max> part being ignored and preserving the pre-3.12 behavior of basing policies on <min>.

**Comment by MichaelRWolf** [comment]
Over 2 months ago, the 1 failing check contains 2 failed tests due to timeout.

Given the number of projects that are stumbling due to this, could someone please rerun the checks, and if the timeout clears, please merge this PR?  !!!

Thanks!

**Review by MichaelRWolf — COMMENTED** [review]
This _minor_ change looks good to me.  It will certainly quiet many deprecation warnings that have been clogging many screens for over 2 months.  Anyone up for closing this by Christmas?

**Comment by Holzhaus** [comment]
Whohoo!

**Comment by mojca** [comment]
Would you be willing to cherry-pick this change into the release branch (v1.10.x)? Thanks in advance.

**Comment by friendlyanon** [comment]
README says:
> Post 1.10.x googletest will follow [Abseil Live at Head philosophy](https://abseil.io/about/philosophy)

So I don't think there will be any more releases.  
My solution is to just fix the download at the commit this was merged.

**Comment by Be-ing** [comment]
> Post 1.10.x googletest will follow Abseil Live at Head philosophy

Considering this critical bug fix took 3 months after reporting it, took an outside contributor to fix, and wasn't merged for another 2 months, all this despite many users speaking up, this doesn't inspire sufficient confidence to me for a "Live at Head" policy.

**Comment by friendlyanon** [comment]
It's to be [expected](https://killedbygoogle.com/) things end up this way with Google.

I can recommend [doctest](https://github.com/onqtam/doctest) as an alternative.

**Comment by Be-ing** [comment]
Thanks for the perspective. Just because a large company puts resources into a project does not mean it is well maintained or will be in the future...

**Comment by mojca** [comment]
I just started testing doctest a few days ago and it seems awesome indeed. The author fixed a bug I reported in less than a day despite not having access to the compiler in question.

The only thing I'm missing a bit at the moment is a proper test adapter for VS (but it works well in VS Code).

**Comment by lesomnus** [comment]
doctesg is good. but google/benchmark uses googletest.

**Comment by LukeAI** [comment]
Is there any suggestion as to when we might see this in a release?

**Comment by patrick96** [comment]
@LukeAI gtest doesn't have releases anymore. It's "live at head": https://github.com/google/googletest#live-at-head

**Pull request merged**